### ISSUE TITLE
VoiceOver/iOS: Anchors do not fire tap

### ISSUE THREAD

**Issue by cdata** [Polymer/polymer#3957]
### Description

Due to an apparent bug in either VoiceOver or WebKit, anchors on iOS (when VoiceOver is enabled) do not fire all expected touch events. **This results in Polymer never firing `tap` events for anchors in this configuration.**

#### Live Demo
http://output.jsbin.com/mapipum

#### Steps to Reproduce
1. Enable VoiceOver on an iOS device
2. Open the demo link in Chrome or Safari
3. Tap the link so that two sequences of input events are printed

#### Expected Results
The following sequence of input events is expected:

```
a#normal: touchstart
a#normal: touchend
a#normal: mousedown
a#normal: mouseup
a#normal: click
a#normal: touchstart
a#normal: touchend
a#normal: mousedown
a#normal: mouseup
a#normal: click
```

#### Actual Results
The `touchstart` event only fires once, and `touchend` never fires:

```
a#normal: touchstart
a#normal: mousedown
a#normal: mouseup
a#normal: click
a#normal: mousedown
a#normal: mouseup
a#normal: click
```
### Browsers Affected

- Chrome/iOS
- Safari 9/iOS

**Comment by cdata** [comment]
I will be filing a bug against WebKit after I have tested this in iOS10.

**Comment by TimvdLippe** [comment]
I think this is a duplicate of #3952, but this issue has a reprocase and a more clearer description

**Comment by jfrazzano** [comment]
Hey Tim, 

Frankie Fu and I also discussed the issue in the context of the Shop app.  Don’t recall the issue number, but it was an iPhone button on tap, that needed a “long hold” to secure the link.  Frankie, I believe, just used a click, but its better to ask him.  Hope that helps.  Not sure its the same issue.  Sorry if its not.

Sincerely, 

Jason

**Comment by cdata** [comment]
A couple of reported (but unverified by me) work-arounds, for anyone who runs up against this:

 1. Set `role="button"` on the anchor tag. This causes all expected events to fire.
 2. Don't add the `href` attribute to the anchor right away. Once `touchstart` fires for the first time, add the `href` attribute back to the anchor. All expected events will fire from then on.

**Comment by cdata** [comment]
I have tested and confirmed that Safari/iOS10 **does not** exhibit the above bug when VoiceOver is enabled.

**Comment by cdata** [comment]
Chrome/iOS 10 **does not** exhibit the above bug when VoiceOver is enabled.

**Comment by azakus** [comment]
So, I can make this work correctly by making `click` always trigger `tap` on touch only devices.
However, this will make a [gray tap highlight](https://developer.apple.com/library/content/documentation/AppleApplications/Reference/SafariWebContent/AdjustingtheTextSize/AdjustingtheTextSize.html#//apple_ref/doc/uid/TP40006510-SW5) appear on iOS devices.
This is controllable with [-webkit-tap-highlight-color](https://developer.mozilla.org/en-US/docs/Web/CSS/-webkit-tap-highlight-color) css property, but there is a question of whether Polymer should disable the tap highlight by default.

@cdata Thoughts?

**Comment by alexclarkofficial** [comment]
I recently discussed iOS gray tap highlight with the accessibility team at my work and they had requested that we do not do anything to remove the gray border. Would it be possible to implement something like this under a feature flag?

I believe this will also solve the issue of a lack of retargeting a `tap` event, the same way a `click` is retargeted for an effectively bigger user input target.

One last question: is there a reason to only add back `click` and not all mouse events?

PS: I believe this will also expose a bug where the `mouseCanceller` is not removed until the timeout.
